osconfig_tests/test_suites/package_management/package_management.go

```diff
@@ -112,44 +108,197 @@ func TestSuite(ctx context.Context, tswg *sync.WaitGroup, testSuites chan *junit
+
+	// allow agent to make the lookupconfig call and install the package
+	time.Sleep(1 * time.Minute)
+	// TODO refactor to remove hardcoding of package name
+	listPkgCmd := getDebListCmd()
+	sshCmd := getGcloudSshCmd(testZone, testSetup.name, listPkgCmd, logger)
+	out, err := run(sshCmd, logger)
+
+	if err != nil {
+		testCase.WriteFailure("Error running verification command: %v", err)
+		return
+	}
+	_ = out
+
+	if err = inst.WaitForSerialOutput("cowsay", 1, 5*time.Second, 5*time.Minute); err != nil {
+		testCase.WriteFailure("Error waiting for assertion: %v", err)
+		return
+	}
+
+}
+
+func getGcloudSshCmd(zone string, instance string, pkgManagerCommand string, logger *log.Logger) *exec.Cmd {
+	return exec.Command("gcloud", []string{"compute", "ssh", "--zone", fmt.Sprintf("%s", zone), instance, "--command", fmt.Sprintf("%s %s\n", pkgManagerCommand, writeToSerialConsole)}...)
+}
+
+//TODO move this to common library
+func run(cmd *exec.Cmd, logger *log.Logger) ([]byte, error) {
+	logger.Printf("Running %q with args %q\n", cmd.Path, cmd.Args[1:])
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return nil, fmt.Errorf("error running %q with args %q: %v, stdout: %s", cmd.Path, cmd.Args, err, out)
+	}
+	return out, nil
+}
+
 func packageManagementTestCase(ctx context.Context, testSetup *packageManagementTestSetup, tests chan *junitxml.TestCase, wg *sync.WaitGroup, logger *log.Logger, regex *regexp.Regexp) {
 	defer wg.Done()
 
 	createOsConfigTest := junitxml.NewTestCase(testSuiteName, fmt.Sprintf("[CreateOsConfig] Create OsConfig"))
+	packageInstallTest := junitxml.NewTestCase(testSuiteName, fmt.Sprintf("[PackageInstall] Pacakge installation"))
 
-	for tc, f := range map[*junitxml.TestCase]func(context.Context, *junitxml.TestCase, *log.Logger){
+	for tc, f := range map[*junitxml.TestCase]func(context.Context, *junitxml.TestCase, *packageManagementTestSetup, *log.Logger){
 		createOsConfigTest: runCreateOsConfigTest,
+		packageInstallTest: runPackageInstallTest,
 	} {
 		if tc.FilterTestCase(regex) {
 			tc.Finish(tests)
 		} else {
 			logger.Printf("Running TestCase %s.%q", tc.Classname, tc.Name)
-			f(ctx, tc, logger)
+			f(ctx, tc, testSetup, logger)
 			tc.Finish(tests)
 			logger.Printf("TestCase %s.%q finished in %fs", tc.Classname, tc.Name, tc.Time)
 		}
 	}
 }
+
+func cleanuposconfig(ctx context.Context, testCase *junitxml.TestCase, logger *log.Logger, oc *osconfigserver.OsConfig) {
+	err := oc.Cleanup(ctx, logger)
+	if err != nil {
+		testCase.WriteFailure(fmt.Sprintf("error while deleting osconfig: %s", err))
+	}
+}
+
+func cleanupassignment(ctx context.Context, testCase *junitxml.TestCase, logger *log.Logger, assignment *osconfigserver.Assignment) {
+	err := assignment.Cleanup(ctx, logger)
+	if err != nil {
+		testCase.WriteFailure(fmt.Sprintf("error while deleting assignment: %s", err))
+	}
+}
+
+func getDebListCmd() string {
+	cmd := dpkgquery
+	for _, arg := range dpkgqueryArgs {
+		cmd = fmt.Sprintf("%s %s", cmd, arg)
+	}
```

Comment:
You could replace this with `return fmt.Sprintf("%s %s", dpkgquery, strings.Join(dpkgqueryArgs, " "))`. Though if this is the only place you use these args, it might be clearer to substitute `dpkgquery` and `dpkgqueryArgs` for a const `debListCmd := "/usr/bin/dpkg-query -W"` and remove this function.

Reply:
oh.. good pick. I earlier had few other arguments, like -f for formatting output, for dpkgquery but eventually realized that i only need -W but forgot to change the method.
will change it to a single var to avoid extra processing.

during the whole experiment with package manager commands, i realized that there is need for a library that can provide standard APIs to perform package manager operations. Otherwise, we might end up writing the same code at multiple places.
I will discuss with @adjackura about it.